virttest/remote_commander/remote_runner.py

```diff
@@ -295,6 +297,10 @@ def parse_func_name(self, func_name, commander):
         :param func_name: Function name
         :param commander: Where to execute the command (remote or local)
         """
+        for name in func_name:
+            if name == "run_test":
+                # Always run external test in the current process.
+                self.manage = True
```

Comment:
Basically this overrides if the user uses `commander.async.run_test` and forces it to use the main thread, because the current implementation does not support async execution, right?

Isn't there a better place to do this? For example I'd put an exception it into the commander's `__getattr__` or the ` CmdEncapsulation` saying that `run_test` is only supported inside the master (or even better add the id handling and make it work inside async :wink:).